hawc/apps/epiv2/managers.py

```diff
@@ -193,6 +194,148 @@ def complete(self):
             "factors",
         ).prefetch_related("design__countries")
 
+    def complete_df(self) -> pd.DataFrame:
+        mapping = {
+            # study
+            **study_df_mapping("study-", "design__study__"),
+            # design
+            "design-pk": "design__pk",
+            # "design-url": "design__get_absolute_url", #todo make a sql function or remove?
```

Comment:
If we want to include this and build it in our SQL query, [CONCAT](https://docs.djangoproject.com/en/4.2/ref/models/database-functions/#concat) is probably our best best.

Reply:
Maybe a function that could be called like `sql_format("/path/{}/{}/","field1","field2")`, where the first argument is a format string, and subsequent arguments are the fields to use in those placeholders?

Reply:
I like that idea and the name.